.github/workflows/tests.yml

```diff
@@ -0,0 +1,43 @@
+name: Open Source Programs Run Tests
+
+on: [push, pull_request]
```

Comment:
Is this necessary?? Maybe the actions will work without this line too

Reply:
Being explicit is also good. Like by having this explicitly on the workflow, we don't need to think of the default github actions options. but good point

Reply:
> Is this necessary?? Maybe the actions will work without this line too

@devkapilbansal if we remove this line then how GitHub will know about what event they have to trigger the workflow I think it doesn't let it be a default.